File: pinyougouparent/pinyougou-manager-web/src/main/java/com/shf/pyg/manager/controller/BrandController.java
package com.shf.pyg.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.shf.pyg.entity.PageResult;
import com.shf.pyg.entity.Result;
import com.shf.pyg.pojo.TbBrand;
import com.shf.pyg.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * 品牌控制器层
 */
@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    /**
     * 查询品牌列表
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    /**
     * 分页
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page,int rows){
        return brandService.findPage(page, rows);
    }

    /**
     * 增加品牌
     */
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try {
            brandService.add(brand);
            return new Result(true, "添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "添加失败");
        }
    }

    /**
     * 根据ID查询实体
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    /**
     * 修改
     * @param brand
     */
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    /**
     * 批量删除
     */
    @RequestMapping("/delete")
    public Result delete(Long []ids){
        try {
            brandService.delete(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    /**
     * 条件分页
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,int page,int rows){
        return brandService.findPage(brand, page, rows);
    }
}

File: pinyougouparent/pinyougou-manager-web/src/main/webapp/js/controller/brand_Controller.js
// 构建模块的前端控制器
app.controller("brandController",function($scope,$controller,brandService){
    $controller("baseController",{$scope:$scope});

    $scope.findAll = function() {
        brandService.findAll().success(function(response){
            $scope.list = response
        })
    }

    // 分页
    $scope.findPage = function(page,rows) {
        brandService.findPage(page,rows).success(function(response){
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    }

    // 新增品牌数据
    $scope.add = function() {
        var object = null;
        if ($scope.entity.id != null){
            object = brandService.update($scope.entity)
        } else {
            objet = brandService.add($scope.entity)
        }
        object.success(function(response){
            if (response.success){
                $scope.reloadList()
            } else {
                alert(response.message)
            }
        })
    }

    $scope.findOne = function(id) {
        brandService.findOne(id).success(function(response) {
            $scope.entity = response
        })
    }


    $scope.dele = function() {
        brandService.dele($scope.selectIds).success(function(response){
            if (response.success){
                // 刷新列表
                $scope.reloadList()
                $scope.selectIds=[]
            } else {
                alert(response.message)
            }
        })
    }

    // 条件查询+分页
    $scope.search = function(page,rows) {
        brandService.search(page,rows,$scope.searchEntity).success(function(response){
            $scope.list = response.rows;
            $scope.paginationConf.totalItems=response.total;
        })
    }
})